model/patch_lifecycle.go

```diff
@@ -425,9 +426,156 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 	if err = p.SetActivated(patchVersion.Id); err != nil {
 		return nil, errors.WithStack(err)
 	}
+
+	if err = processTriggerAliases(ctx, p, projectRef); err != nil {
+		return nil, errors.Wrap(err, "problem processing trigger aliases")
+	}
+
 	return patchVersion, nil
 }
 
+func processTriggerAliases(ctx context.Context, p *patch.Patch, projectRef *ProjectRef) error {
+	if len(p.Triggers.Aliases) == 0 {
+		return nil
+	}
+
+	type aliasGroup struct {
+		project        string
+		status         string
+		parentAsModule string
+	}
+	aliasGroups := make(map[aliasGroup][]PatchTriggerDefinition)
+	for _, aliasName := range p.Triggers.Aliases {
+		aliasFound := false
+		var alias PatchTriggerDefinition
+		for _, alias = range projectRef.PatchTriggerAliases {
+			if alias.Alias == aliasName {
+				aliasFound = true
+				break
+			}
+		}
+		if !aliasFound {
+			// the user specified a non-existent alias
+			continue
+		}
+
+		childProjectID, err := FindIdForProject(alias.ChildProject)
+		if err != nil {
+			return errors.Wrapf(err, "can't get ID for child project '%s'", alias.ChildProject)
+		}
+
+		// group patches on project, status, parentAsModule
+		group := aliasGroup{
+			project:        childProjectID,
+			status:         alias.Status,
+			parentAsModule: alias.ParentAsModule,
+		}
+		aliasGroups[group] = append(aliasGroups[group], alias)
+	}
+
+	childPatchIDs := []string{}
+	for group, definitions := range aliasGroups {
+		if group.status == "" {
+			childPatchID, err := makePatchFromDefinitions(ctx, p, group.project, group.parentAsModule, definitions)
+			if err != nil {
+				return errors.Wrap(err, "problem creating patch for alias definitions")
+			}
+			if childPatchID != "" {
+				childPatchIDs = append(childPatchIDs, childPatchID)
+			}
+		} else {
+			if err := subscribeOnPatchOutcome(p, group.status, definitions); err != nil {
+				return errors.Wrap(err, "problem subscribing alias definitions on patch outcome")
+			}
+
+		}
+	}
+
+	return p.AppendChildPatches(childPatchIDs)
+}
+
+func makePatchFromDefinitions(ctx context.Context, p *patch.Patch, projectID, parentAsModule string, definitions []PatchTriggerDefinition) (string, error) {
+	if len(definitions) == 0 {
+		return "", nil
+	}
+
+	v, project, err := FindLatestVersionWithValidProject(projectID)
```

Comment:
Would it make more sense to have this instead be the base version for the patch?

Reply:
My thinking here was that we need to get a base version for the child patch.